LoginAndAclSetup.md

```diff
@@ -0,0 +1,178 @@
+# Setting up optional login and document level access control
+
+## Table of Contents
+
+- [Requirements](#requirements)
+- [Setting up Azure AD Apps](#setting-up-azure-ad-apps)
+  - [Manual Setup](#manual-setup)
+    - [Server App](#server-app)
+    - [Client App](#client-app)
+- [Optional Scripts](#optional-scripts)
+  - [Azure Data Lake Storage Gen2 Setup](#azure-data-lake-storage-gen2-setup)
+  - [Azure Data Lake Storage Gen2 Prep Docs](#azure-data-lake-storage-gen2-prep-docs)
+  - [Manually managing Document Level Access Control](#manually-managing-document-level-access-control)
+- [Environment Variables Reference](#environment-variables-reference)
+
+This guide demonstrates how to add an optional login and document level access control system to the sample. This system can be used to restrict access to indexed data to specific users based on what [Azure Active Directory (Azure AD) groups](https://learn.microsoft.com/azure/active-directory/fundamentals/how-to-manage-groups) they are a part of, or their [user object id](https://learn.microsoft.com/partner-center/find-ids-and-domain-names#find-the-user-object-id).
+
+![AppLoginArchitecture](./docs/applogincomponents.png)
+
+## Requirements
+
+**IMPORTANT:** In order to add optional login and document level access control, you'll need the following in addition to the normal sample requirements
+
+* **Azure account permissions**: Your Azure account must have [permission to manage applications in Azure AD](https://learn.microsoft.com/azure/active-directory/develop/quickstart-register-app#prerequisites).
+
+## Setting up Azure AD Apps
+
+Two Azure AD apps must be registered in order to make the optional login and document level access control system work correctly.
+
+### Manual Setup
+
+The following instructions explain how to setup the two apps using the Azure Portal.
+
+#### Setting up the Server App
+
+* Sign in to the [Azure portal](https://portal.azure.com/).
+* Select the Azure AD Service.
+* In the left hand menu, select **Application Registrations**.
+* Select **New Registration**.
+  * In the **Name** section, enter a meaningful application name. This name will be displayed to users of the app, for example `Azure Search OpenAI Demo API`.
+  * Under **Supported account types**, select **Accounts in this organizational directory only**.
+* Select **Register** to create the application
+* In the app's registration screen, find the **Application (client) ID**.
+  * Run the following `azd` command to save this ID: `azd env set AZURE_SERVER_APP_ID <Application (client) ID>`.
+* Select **Certificates & secrets** in the left hand menu.
```

Comment:
It will only be possible in Bicep to use certificates, and the docs all discourage the use of secrets, so I wonder if we should encourage the (admittingly more difficult) route of using a certificate here. Otherwise it'll be hard for folks to upgrade.

https://learn.microsoft.com/en-us/graph/applications-how-to-add-certificate

"Azure Active Directory (Azure AD) supports three types of credentials to authenticate apps and service principals: passwords (app secrets), certificates, and federated identity credentials. If you can't use federated identity credentials for your app, we strongly recommend that you use certificates instead of secrets."

The easiest way to get started locally with certificates is self-signed certificates, though those are also discouraged for production.

Reply:
So this makes it more challenging - I'll see if there's an easy way to add key vault support here. Are you OK leaving in the password for now and creating a follow up issue?

Reply:
Yeah I'm fine with that. I'm very curious how we can empower developers to use certificates, based on my difficulties setting them up.

Reply:
Discussed offline - there is an opportunity to enhance this using Azure Key Vault + self-signed certs once we have bicep support.

Reply:
Added a note stating that certificates are considered more secure and linked to the certificate creation docs